core/vm/logger.go

```diff
@@ -65,36 +58,34 @@ type StructLog struct {
 	Err     error
 }
 
+type VmLogger interface {
+	CaptureState(env Environment, pc uint64, op OpCode, gas, cost *big.Int, memory *Memory, stack *stack, contract *Contract, depth int, err error);
+}
```

Comment:
I've renamed `Logger` to `StructLogger` and `VmLogger` to `Logger`, giving the interface the base name as you suggested.

I also realised I need to expose Stack in order for this to be usable outside the VM package.